client/sites/marketing/tools/index.tsx

```diff
@@ -85,103 +41,32 @@ export const MarketingTools: FunctionComponent = () => {
-					) }
-					imagePath={ earnIllustration }
-					imageAlt={ translate( 'A stack of coins' ) }
-				>
-					<FeaturedButton onClick={ handleEarnClick }>
-						{ translate( 'Start earning' ) }
-					</FeaturedButton>
-				</MarketingToolsFeature>
-
-				<MarketingToolsFeature
-					title={ translate( 'Fiverr logo maker' ) }
-					description={ translate(
-						'Create a standout brand with a custom logo. Our partner makes it easy and quick to design a professional logo that leaves a lasting impression.'
-					) }
-					imagePath={ fiverrLogo }
-					imageAlt={ translate( 'Fiverr logo' ) }
-				>
-					<FeaturedButton
-						onClick={ handleCreateALogoClick }
-						href="https://wp.me/logo-maker/?utm_campaign=marketing_tab"
-						target="_blank"
-					>
-						{ translate( 'Make your brand' ) }
-					</FeaturedButton>
-				</MarketingToolsFeature>
-
-				<MarketingToolsFeature
-					title={ translate( 'Hire an SEO expert' ) }
-					description={ translate(
-						'In today’s digital age, visibility is key. Hire an SEO expert to boost your online presence and capture valuable opportunities.'
-					) }
-					imagePath={ fiverrLogo }
-					imageAlt={ translate( 'Fiverr logo' ) }
-				>
-					<FeaturedButton
-						onClick={ handleHireAnSEOExpertClick }
-						href="https://wp.me/hire-seo-expert/?utm_source=marketing_tab"
-						target="_blank"
-					>
-						{ translate( 'Talk to an SEO expert today' ) }
-					</FeaturedButton>
-				</MarketingToolsFeature>
-
-				<MarketingToolsFeature
-					title={ translate( 'Get social, and share your blog posts where the people are' ) }
-					description={ translate(
-						"Use your site's Jetpack Social tools to connect your site and your social media accounts, and share your new posts automatically. Connect to Facebook, LinkedIn, and more."
-					) }
-					imagePath="/calypso/images/marketing/social-media-logos.svg"
-					imageAlt={ translate( 'Logos for Facebook, Twitter, LinkedIn, and Tumblr' ) }
-				>
-					<FeaturedButton onClick={ handleStartSharingClick }>
-						{ translate( 'Start sharing' ) }
-					</FeaturedButton>
-				</MarketingToolsFeature>
-
-				{ isEnglish && (
-					<MarketingToolsFeature
-						title={ translate( 'Increase traffic to your WordPress.com site' ) }
-						description={ translate(
-							'Take our free introductory course about search engine optimization (SEO) and learn how to improve your site or blog for both search engines and humans.'
-						) }
-						imagePath={ rocket }
-						imageAlt={ translate( 'A rocketship' ) }
-					>
-						<FeaturedButton
-							onClick={ handleSEOCourseClick }
-							href="https://wordpress.com/learn/courses/intro-to-seo/"
-							target="_blank"
+				{ marketingFeatures.map( ( feature, index ) => {
+					if ( feature.showOnlyInEnglish && ! isEnglish ) {
+						return null;
+					}
```

Comment:
I would move the `featureShouldShow` logic to the data as a function that accepts `props` and returns a boolean. Then it's feature can build their own `featureShouldShow` function. Most of them will just return `true` while some of them might have some logic like `showOnlyInEnglish`. It might sound like over-engineering but the other day we wanted to show a feature only on specific days.

Reply:
After looking at the code, I agree that business logic belongs to the data. That way, if further logic is required in the future for another item, it can be implemented in the code that populates the list and the parent component that renders the list will not be modified. 

I have implemented the change in [94f4cdd](https://github.com/Automattic/wp-calypso/pull/96542/commits/94f4cddc9279e18071dd8d977ebc4638679f1be3), I have explicitly added the extra item in an additional step as I think is easier to read. For easier review, I recommend to `Hide whitespace` 
<img src="https://github.com/user-attachments/assets/38fb634d-9c5c-4de5-95f3-30bdcaee7d8c" width=150 />

Thanks for the great suggestion!

Reply:
As discussed, I have move the additional logic to the _data_ component here [62f148e](https://github.com/Automattic/wp-calypso/pull/96542/commits/62f148e18dfe91802e43550b50fe23f4c2f9eac0)